From 64afb8a930b3bf7a02013e0655085275862f4efc Mon Sep 17 00:00:00 2001
From: zhang xingxing <2388969553@qq.com>
Date: Tue, 13 Oct 2026 06:59:47 +0200
Subject: [PATCH] avcodec/cbs_apv: reject user defined metadata shorter than
 its UUID

This is not a security issue.

Fixes: 00smV7L5DOfY
Fixes: swarm-metadata_user_defined-data_size-underflows-on-payload_size-16-5ff8a8

Found-by: zhang xingxing <2388969553@qq.com>
Signed-off-by: Michael Niedermayer <michael@niedermayer.cc>
(cherry picked from commit 64afb8a930b3bf7a02013e0655085275862f4efc)
---
 libavcodec/cbs_apv_syntax_template.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/libavcodec/cbs_apv_syntax_template.c b/libavcodec/cbs_apv_syntax_template.c
index 8640bcfa003..3cdb320ca4b 100644
--- a/libavcodec/cbs_apv_syntax_template.c
+++ b/libavcodec/cbs_apv_syntax_template.c
@@ -441,6 +441,9 @@ static int FUNC(metadata_user_defined)(CodedBitstreamContext *ctx,
 
     HEADER("User-Defined Metadata");
 
+    if (payload_size < 16)
+        return AVERROR_INVALIDDATA;
+
     for (int i = 0; i < 16; i++)
         ubs(8, uuid[i], 1, i);
 
-- 
2.53.0

